tutorial/adding-progressive-image-loading.md

```diff
@@ -37,98 +37,98 @@ Go to the file which uses `ResponsiveImage` imported from `@quintype/components`
-  ...
-  return (
-    <div className="story-grid">
-      {stories.map(story => (
-        <StoryGridStoryItem story={story} />
-      ))}
-    </div>
-  );
-}
-
-```
+   ```javascript
+   const [perfObj, setPerfObj] = useState({size: '5vw', blur: 10});
+
+   useEffect(() => {
+     setTimeout(() => {
+       setPerfObj({size: '25vw', blur: 0});
+     }, 2500);
+   }, []);
+   ```
+
+3. Pass the size and blur values to the `ResponsiveImage` component. Your final code should look something like this.
+
+   ```javascript
+   ...
+   import React, { useState, useEffect } from "react";
+   import { Link, ResponsiveImage } from "@quintype/components";
+   ...
+   ...
+
+   function StoryGridStoryItem(props) {
+     const [perfObj, setPerfObj] = useState({ size: "5vw", blur: 10 });
+
+     useEffect(() => {
+       setTimeout(() => {
+         setPerfObj({ size: "25vw", blur: 0 });
+       }, 2500);
+     }, []);
+
+     return (
+       <Link href={`/${props.story.slug}`} className="story-grid-item">
+         <figure className="qt-image-16x9" styleName="story-grid-item-image">
+           <ResponsiveImage
+             slug={props.story["hero-image-s3-key"]}
+             metadata={props.story["hero-image-metadata"]}
+             aspectRatio={[16, 9]}
+             defaultWidth={480}
+             widths={[250, 480, 640]}
+             sizes={perfObj.size}
+             imgParams={{ auto: ["format", "compress"], blur:perfObj.blur }}
+             eager={props.position < 2 ? "above-fold" : "below-fold"}
+             alt={props.story.headline || ""}
+           />
+         </figure>
+         ...
+         ...
+       </Link>
+     );
+   }
+
+   export function StoryGrid({ stories = [] }) {
+     ...
+     return (
+       <div className="story-grid">
+         {stories.map(story => (
+           <StoryGridStoryItem story={story} />
+         ))}
+       </div>
+     );
+   }
+
+   ```
 
 **Note:**
 
 1. The progressive image loading makes two calls for a particular image, during page load and after a specified interval.
-2. Progressive image loading should be applied for the rows above the fold or in the initial viewport on each page.
+2. Progressive image loading should be applied for the rows above the fold or in the initial viewport on each page to avoid multiple calls for a particular.
 3. For the pages which consist of only one row (Ex: Components with Load More button), this feature would be applied for all the stories which are loaded initially.
 
 ### How to achieve progressive image loading for the rows/components above the fold ?
 
 - For Home Page/Collection Page/Section Page:
 
-  - Utilize the index which will be available in each row.
+  - Home or collection page utilizes either `Collection` or `LazyCollection` HOC to render the rows/components. These HOC components return the index to each of its rows. Utilize this index value to achieve progressive image loading for the rows/components above the fold.
```

Comment:
can rephrase `These HOC components return the index` to `These higher order components return the index`

Reply:
sure